streams/src/test/java/org/apache/kafka/streams/kstream/internals/KTableFilterTest.java

```diff
@@ -97,7 +99,8 @@ public void testQueryableKTable() {
 
         final KTable<String, Integer> table1 = builder.table(topic1, consumed);
 
-        final KTable<String, Integer> table2 = table1.filter(predicate, Materialized.as("anyStoreNameFilter"));
+        final KTable<String, Integer> table2 = table1.filter(predicate,
```

Comment:
This is needed since with the optimization, we will omit flush those unnecessary nulls.

Reply:
Why not update the expected test output? Or better, have a test for both cases?

Reply:
This should be covered in the store test case itself, not the KTableFilter here. I've added a separate test case for this purpose.

Reply:
Well. Guess the "problem" is, that it's unclear what this test is supposed to test. The method name is not very good.

It seems to be about the case, that the filter does the right thing based on the input data. 
 - `testKTable` -- table store is not materialized thus everything is pass-through
 - `testQueryableKTable` -- table store is materialized (we disable caching to preserve pass-through behavior)
similar below for
 - `testNotSendingOldValue` (not materialized)
 - `testQueryableNotSendingOldValue` (preserve pass-through via disabling caching)

Does this sound correct?

Reply:
This sounds right to me. Let me rename the test cases a bit to be more clear.